File: leetcode-04/solutions/python3/0187.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        ans = set()
        set_ = set()

        for i in range(len(s) - 9):
            seq = s[i:i + 10]
            if seq in set_:
                ans.add(seq)
            set_.add(seq)

        return list(ans)

File: leetcode-04/solutions/cpp/0187.cpp
class Solution {
 public:
  vector<string> findRepeatedDnaSequences(string s) {
    const int n = s.length();

    unordered_set<string> ans;
    unordered_set<string> set;

    for (int i = 0; i <= n - 10; i++) {
      string seq = s.substr(i, 10);
      if (set.count(seq)) ans.insert(seq);
      set.insert(seq);
    }

    return vector<string>(ans.begin(), ans.end());
  }
};